lib/src/tasks/test/cli.dart

```diff
@@ -104,20 +80,37 @@ class TestCli extends TaskCli {
     List<String> platforms =
         TaskCli.valueOf('platform', parsedArgs, config.test.platforms);
 
-    if (hasRestParams(parsedArgs)) {
-      individualTests = await addToTestsFromRest(tests, parsedArgs.rest);
-    }
+    // CLI user can specify individual test files to run.
+    bool individualTestsSpecified = parsedArgs.rest.isNotEmpty;
+
+    // The unit test suite should be run by default.
+    bool unit = parsedArgs['unit'] ?? true;
 
-    if (isExplicitlyFalse(unit) && !integration && individualTests == 0) {
+    // The integration suite should only be run if the --integration is set.
+    bool integration = parsedArgs['integration'] ?? false;
+
+    // CLI user can filter tests by name.
+    bool testNamed = parsedArgs['name'] != null;
+
+    if (!individualTestsSpecified && !unit && !integration) {
       return new CliResult.fail(
-          'No tests were selected. Include at least one of --unit or --integration.');
+          'No tests were selected. Include at least one of --unit or '
+          '--integration, or pass in one or more test files/directories');
     }
 
-    if (unit) {
-      tests.addAll(config.test.unitTests);
-    }
-    if (integration) {
-      tests.addAll(config.test.integrationTests);
+    // Build the list of tests to run.
+    if (individualTestsSpecified) {
+      // Individual tests explicitly passed in should override the test suites.
+      tests.addAll(parsedArgs.rest);
+    } else {
+      // Unit and/or integration suites should only run if individual tests
+      // were not specified.
+      if (unit) {
+        tests.addAll(config.test.unitTests);
+      }
+      if (integration) {
+        tests.addAll(config.test.integrationTests);
+      }
     }
 
     if (tests.isEmpty) {
```

Comment:
do we want to move this conditional up into the block above because currently you can see a failure if you try to run a named test but don't have unit tests specified in your dev.dart file